zuul-core/src/main/java/com/netflix/zuul/message/http/HttpResponseMessageImpl.java

```diff
@@ -179,24 +179,19 @@ public int getMaxBodySize() {
     @Override
     public Cookies parseSetCookieHeader(String setCookieValue) {
         Cookies cookies = new Cookies();
-        for (Cookie cookie : CookieDecoder.decode(setCookieValue)) {
-            cookies.add(cookie);
-        }
+        cookies.add(ClientCookieDecoder.STRICT.decode(setCookieValue));
         return cookies;
     }
 
     @Override
     public boolean hasSetCookieWithName(String cookieName) {
-        boolean has = false;
         for (String setCookieValue : getHeaders().getAll(HttpHeaderNames.SET_COOKIE)) {
-            for (Cookie cookie : CookieDecoder.decode(setCookieValue)) {
-                if (cookie.getName().equalsIgnoreCase(cookieName)) {
-                    has = true;
-                    break;
-                }
+            Cookie cookie = ClientCookieDecoder.STRICT.decode(setCookieValue);
```

Comment:
Intentional to use STRICT here vs LAX in `HttpRequestMessageImpl.java` and further down in add/setCookie?

Reply:
Yeah, I maintained the strict/lax behavior. By default CookieDecoder.decode() uses strict, but for whatever reason ServerCookieEncoder.encode uses lax